### PR TITLE
Add barcode scanner to Shrine example

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

Add a live barcode scanner to the Shrine example in the flutter_gallery app.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

## Tests

No tests since this is uses camera and firebase_ml_vision. Also, its example code.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: examples/flutter_gallery/ios/Runner/Info.plist [hunk 1]
```diff
@@ -43,5 +43,9 @@
 	<false/>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<false/>
+	<key>NSCameraUsageDescription</key>
+  <string>Can we use the camera to scan barcodes?</string>
```
Comment (skreborn) [inline examples/flutter_gallery/ios/Runner/Info.plist]: This looks like (accidental) inconsistent spacing. All other lines use tabs, except for these 3.

File: examples/flutter_gallery/lib/demo/shrine/backdrop.dart [hunk 2]
```diff
@@ -66,9 +68,9 @@ class _BackdropTitle extends AnimatedWidget {
     this.onPress,
     @required this.frontTitle,
     @required this.backTitle,
-  }) : assert(frontTitle != null),
-       assert(backTitle != null),
-       super(key: key, listenable: listenable);
+  })  : assert(frontTitle != null),
```
Comment (skreborn) [inline examples/flutter_gallery/lib/demo/shrine/backdrop.dart]: Why the extra padding?

**Comment by KirolousNashaat** [comment]
related : https://github.com/flutter/flutter/issues/32526
btw i tried your code but there is something wrong with barcode.boundingbox

its x,y are reversed for some reason
so i need to do this to fix the check if the barcode is in valid:

`final reversedXY = Rect.fromLTRB(
          barcode.boundingBox.top, barcode.boundingBox.left, barcode.boundingBox.bottom, barcode.boundingBox.right);
      final Rect intersection = validRect.intersect(reversedXY);
`

**Comment by skreborn** [comment]
@KirolousNashaat I believe this might be because [the detector is given the camera's current orientation](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R190), but I don't see [`_handleResult`](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R208) accounting for the same. Perhaps your camera is detected as rotated - can you test and see what [`camera.sensorOrientation`](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R74) is for you.

**Comment by KirolousNashaat** [comment]
> @KirolousNashaat I believe this might be because [the detector is given the camera's current orientation](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R190), but I don't see [`_handleResult`](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R208) accounting for the same. Perhaps your camera is detected as rotated - can you test and see what [`camera.sensorOrientation`](https://github.com/flutter/flutter/pull/32646/files#diff-452a2b7c07fab642c2990d2036b11833R74) is for you.
> 
> Then again, I might be completely off here.

u are right , just tested my orientation is 90 in portrait ? 
i think i should rotate the validRect with my sensor orientation , 0 / 90
i don't think i need to do 180/270 right ?

**Comment by KirolousNashaat** [comment]
like this : 
`if (cameraDescription.sensorOrientation == 90 || cameraDescription.sensorOrientation == 270) {
    validRect = Rect.fromLTRB(
        center.dy - halfHeight,
        center.dx - halfWidth,
        center.dy + halfHeight,
        center.dx + halfWidth,
    );
} else {
    validRect = Rect.fromLTRB(
        center.dx - halfWidth,
        center.dy - halfHeight,
        center.dx + halfWidth,
        center.dy + halfHeight,
    );
}`

btw : 
`_cameraController.stopImageStream();` freezes the app for a moment

**Comment by bparrishMines** [comment]
This example is now in the flutter/plugins repo with PR https://github.com/flutter/plugins/pull/1758

**Pull request closed without merging**